File: src/main/java/Pages/LoginPage.java
package Pages;

import Management.Info;
import Management.PageObject;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.FindBys;
import org.openqa.selenium.support.How;
import org.testng.Assert;

import java.util.List;

/**
 * Created by valeriy on 4/1/2017.
 */
public class LoginPage extends PageObject {
    Info info = new Info();
    @FindBy(xpath = "//h1[@class='page-title']")
    private WebElement authorization;
    @FindBy(xpath = "//button[@class='btn-input btn-input-block']")
    private WebElement eye;
    @FindBy(name = "email")
    private WebElement emailField;
    @FindBy(name = "password")
    private WebElement passwordField;
    @FindBy(xpath = "//button[@type='submit']")
    private WebElement loginButton;
    @FindBy(how = How.CLASS_NAME, using = "noty_text")
    private WebElement notification;
    @FindBy(how = How.CLASS_NAME, using = "tooltip-text")
    private WebElement emailErrorTooltip;
    @FindBys(@FindBy(className = "tooltip-text"))
    List<WebElement> toolTips;

    public LoginPage(WebDriver driver) {
        super(driver);
        Assert.assertTrue(authorization.isDisplayed());
    }

    public void clickOnEye() {
        eye.click();
    }

    public void enterLogin(String login) {
        emailField.sendKeys(login);
    }

    public void enterPassword(String password) {
        passwordField.sendKeys(password);
    }

    public HomePage clickLoginButton() {
        loginButton.click();
        return new HomePage(driver);
    }

    public boolean clickLoginButtonForNotification() {
        loginButton.click();
        try {
            Thread.sleep(2000);
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return notification.isDisplayed() ? true : false;
    }

    public boolean checkNotificationText(String expectedText) {
        return notification.getText().equals(expectedText);
    }

    public void clickLoginButtonNoAction() {
        loginButton.click();
    }

    public boolean checkToolTipText(int toolTipNumber, String expectedText) {
        return toolTips.get(toolTipNumber).getText().equals(expectedText) ? true : false;
    }
    public void signIn(String login, String password){
        enterLogin(login);
        enterPassword(password);
    }
    public String getEnteredPassword(){
        clickOnEye();
        return passwordField.getAttribute("value");
    }
}

File: PageObjects/LoginPage.js
let BasePage = require('./BasePage');

class LoginPage extends BasePage {
  constructor() {
    super();
    this.logInButton = element(by.xpath('//a[text()="Log in"]'));
    this.emailField = element(by.xpath('//div[@class ="form-group email"]/child::input[contains(@name,"email")]'));
    this.passwordField = element(by.xpath('//input[contains(@name,"password")]'));
    this.showPassword = element(by.xpath('//button[contains(@class,"btn-input")]'));
    this.login = element(by.xpath('//button[text()="Login"]'));
    this.url = 'https://ssls.com';
    this.logText = element(by.xpath('//div[@class="log-box"]'));
    this.notifyMessage = element(by.xpath('//span[@class="noty_text"]'));
    this.leftSideTooltip = element(
      by.xpath('//input[@name="email"]/following-sibling::div[@class="left-tooltip-box"]')
    );
    this.emptyFieldsValidationError = element(
      by.xpath('//div[@class="input-group"]/following-sibling::div[@class="left-tooltip-box"]')
    );
    this.logOutButton = element(by.xpath('//button[text()="Log out"]'));
  }

  clickLogInButton() {
    this.login.click();
  }

  clickOnEyeButton() {
    this.showPassword.click();
  }

  async logOut() {
    await this.waitForElementVisibility(this.dropDownMenu);
    this.dropDownMenu.click();
    await this.waitForElementVisibility(this.logOutButton);
    this.logOutButton.click();
  }

  fillInlogInForm(email, password) {
    this.logInButton.click();
    this.emailField.sendKeys(email);
    this.passwordField.sendKeys(password);
  }
}

module.exports = new LoginPage();
